src/main/Creation.java

```diff
@@ -0,0 +1,159 @@
+     * @param resourceGroupName Resource Group of ANF resource
+     * @param anfAccountName ANF account name
+     * @param capacityPoolName ANF capacity Pool name
+     * @param serviceLevel ANF capacity pool service level {Premium or Standard}
+     * @param location ANF resource location / region
+     * @return
+     */
+    public static CompletableFuture<Observable<CapacityPoolInner>> createOrUpdateCapacityPoolAsync(AzureNetAppFilesManagementClientImpl anfClient, String resourceGroupName, String anfAccountName, String capacityPoolName, String serviceLevel, long capacityPoolSize, String location)
+    {
+        CapacityPoolInner capacityPool = new CapacityPoolInner();
+        capacityPool.withServiceLevel(ServiceLevel.fromString(serviceLevel));
+        capacityPool.withSize(capacityPoolSize);
+        capacityPool.withLocation(location);
+
+        Observable<CapacityPoolInner> capacityPoolObservable = anfClient.pools().createOrUpdateAsync(resourceGroupName, anfAccountName, capacityPoolName, capacityPool);
+        return CompletableFuture.completedFuture(capacityPoolObservable);
+    }
+
+    /**
+     * Creates or updates ANF volume resource
+     * @param anfClient Azure NetApp Files Management Client
+     * @param resourceGroupName Resource Group of ANF resource
+     * @param anfAccountName ANF account name
+     * @param capacityPoolName ANF capacity Pool name
+     * @param volumeName ANF volume name
+     * @param volumeSize ANF volume size in Byte
+     * @param serviceLevel ANF capacity pool service level {Premium or Standard}
+     * @param subnetId Subnet Id
+     * @param location ANF resource location / region
+     * @return
+     */
+    public static CompletableFuture<Observable<VolumeInner>> createOrUpdateVolumeAsync(AzureNetAppFilesManagementClientImpl anfClient, String resourceGroupName, String anfAccountName,String capacityPoolName, String volumeName, long volumeSize, String serviceLevel, String subnetId, String location)
+    {
+        List<ExportPolicyRule> ruleList = new ArrayList<>();
+        ruleList.add(new ExportPolicyRule()
+        .withAllowedClients("0.0.0.0")
+        .withRuleIndex(1)
+        .withUnixReadWrite(true)
+        .withUnixReadOnly(false)
+        .withCifs(false)
+        .withNfsv3(false)
+        .withNfsv41(true));
+
+        VolumePropertiesExportPolicy exportPolicy = new VolumePropertiesExportPolicy().withRules(ruleList);
+
+        List<String> protocols = new ArrayList<>();
+        protocols.add("NFSv4.1");
+
+        VolumeInner volumeInner = new VolumeInner();
+        volumeInner.withCreationToken(volumeName);
+        volumeInner.withExportPolicy(exportPolicy);
+        volumeInner.withServiceLevel(ServiceLevel.fromString(serviceLevel));
+        volumeInner.withSubnetId(subnetId);
+        volumeInner.withUsageThreshold(volumeSize);
+        volumeInner.withProtocolTypes(protocols);
+        volumeInner.withLocation(location);
```

Comment:
Can we use the same pattern as in line 68?

Reply:
I have done it but for a reason I don't know Java needs to cast to the same type.